src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/SnippetTaglet.java

```diff
@@ -0,0 +1,354 @@
+/*
+ * Copyright (c) 2020, 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.javadoc.internal.doclets.toolkit.taglets;
+
+import com.sun.source.doctree.AttributeTree;
+import com.sun.source.doctree.DocTree;
+import com.sun.source.doctree.SnippetTree;
+import com.sun.source.doctree.TextTree;
+import jdk.javadoc.doclet.Taglet;
+import jdk.javadoc.internal.doclets.toolkit.Content;
+import jdk.javadoc.internal.doclets.toolkit.DocletElement;
+import jdk.javadoc.internal.doclets.toolkit.taglets.snippet.Action;
+import jdk.javadoc.internal.doclets.toolkit.taglets.snippet.ParseException;
+import jdk.javadoc.internal.doclets.toolkit.taglets.snippet.Parser;
+import jdk.javadoc.internal.doclets.toolkit.taglets.snippet.StyledText;
+import jdk.javadoc.internal.doclets.toolkit.util.Utils;
```

Comment:
The langtools/javac/javadoc convention is to put standard JavaSE imports (`java.*`, `javax.*` ) first.

Reply:
Since we are here, is there any langtools convention for when to collapse a span of single-import declarations into an import-on-demand (.*) declaration?

Reply:
Fixed in a7cd155.

Reply:
> Since we are here, is there any langtools convention for when to collapse a span of single-import declarations into an import-on-demand (.*) declaration?

langtools does not use import-on-demand, and it is practice e to expand them to single imports if any creep into the code.   Also, imports are ordered with Java SE (`java.*` `javax.*`) imports first, before others.